### PR TITLE
fix(datetime): if no default value, don't highlight active day until one is selected

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

When an `ion-datetime` is initialized without a value defined, today's date is highlighted with a solid blue circle. This is misleading, as it looks like a day has already been selected.

Issue URL: https://github.com/ionic-team/ionic-framework/issues/24456

## What is the new behavior?

- For datetimes with no initial value, no solid blue circle is rendered at first. Today's date is highlighted with an outline only. Once any day is clicked, the solid circle returns.
- For datetimes with an initial value defined, behavior is unchanged.
- The solid blue circle does _not_ necessarily indicate which day the actual `value` property is set to at that moment. (I.e. it's possible to trigger the solid circle, but still have a nullish `value`.) This is to handle datetimes with confirm/cancel buttons, where the `value` does not change until `confirm` is called. (See: https://ionicframework.com/docs/api/datetime#confirming-selected-values)

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- We decided not to enforce a non-nullish `value` in order to avoid unwanted `ionChange` events that would fire when the value is automatically switched to today.
- An alternative would be to not default the `activeParts` to today if no initial value is set, but that would require allowing `activeParts` or all its properties to be `null`, which would require a small refactor. I figure the current method is simpler and safer.
- Dev build: `6.1.2-dev.11650392645.12c652f6` (Posted in original issue)

### CODE REVIEW

File: core/src/components/datetime/test/basic/e2e.ts [hunk 1]
```diff
@@ -98,6 +98,41 @@ describe('datetime: selecting a day', () => {
 
     expect(newActiveDay.innerText).toEqual('13');
   });
+
+  const testHighlight = async (page) => {
```
Comment (sean-perkins) [inline core/src/components/datetime/test/basic/e2e.ts:102]: Any blockers that prevented writing this test in Playwright?

Comment (amandaesmith3) [reply core/src/components/datetime/test/basic/e2e.ts:102]: ... I just forgot :joy: I'll do that, haha.

File: core/src/components/datetime/test/basic/datetime.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    );
+
+    const classes = await todayBtn.evaluate((el: any) => el.classList.value);
+    expect(classes).toMatch('calendar-day-today');
+    expect(classes).not.toMatch('calendar-day-active');
+
+    await todayBtn.click();
+    await page.waitForChanges();
+
+    const classes2 = await todayBtn.evaluate((el: any) => el.classList.value);
```
Comment (liamdebeasi) [inline core/src/components/datetime/test/basic/datetime.e2e.ts:21]: You should be able to use the `toHaveClass` method instead: https://playwright.dev/docs/test-assertions#locator-assertions-to-have-class

Comment (liamdebeasi) [reply core/src/components/datetime/test/basic/datetime.e2e.ts:21]: Same thing on line 14

File: core/src/components/datetime/datetime.tsx [hunk 3]
```diff
@@ -1424,7 +1427,7 @@ export class Datetime implements ComponentInterface {
                 class={{
                   'calendar-day-padding': day === null,
                   'calendar-day': true,
-                  'calendar-day-active': isActive,
+                  'calendar-day-active': isActive && highlightActiveParts,
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.tsx:1430]: Can we add a comment for why this is needed?

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**